ff/src/fields/arithmetic.rs

```diff
@@ -1,14 +1,9 @@
-/// This modular multiplication algorithm uses Montgomery
-/// reduction for efficient implementation. It also additionally
-/// uses the "no-carry optimization" outlined
-/// [here](https://hackmd.io/@zkteam/modular_multiplication) if
-/// `P::MODULUS` has (a) a non-zero MSB, and (b) at least one
-/// zero bit in the rest of the modulus.
-macro_rules! impl_field_mul_assign {
+macro_rules! impl_no_carry {
     ($limbs:expr) => {
+        // XXX. This function should be made "const" in a glorious future.
```

Comment:
Is the reason it can't be const right now because of indexing? Or because it access P::MODULUS?

Reply:
```error[E0658]: trait bounds other than `Sized` on const fn parameters are unstable```
cf. https://github.com/rust-lang/rust/issues/57563 and the feature gate `const_fn_trait_bound`

Reply:
Ah yeah, so it's because we're inside `impl<P: FpParameters> FpXYZ<P>`. Hopefully we'll see that get stabilized, because then basically half this library becomes `const` immediately...

Reply:
This is the relevant feature: https://github.com/rust-lang/rust/issues/67792

Reply:
Should I remove this line, keep it as a placeholder elsewhere?

Reply:
Nah, it's fine.